code/modules/projectiles/projectile/special.dm

```diff
@@ -43,26 +43,59 @@
 	return 1
 
 /obj/item/projectile/temp
-	name = "freeze beam"
+	name = "temperature beam"
 	icon_state = "ice_2"
 	damage = 0
 	damage_type = BURN
 	nodamage = 1
 	flag = "energy"
-	var/temperature = 100
-
+	temperature = null
+
+/obj/item/projectile/temp/New()
+	spawn(1)

```

Comment:
I don't know, but without that spawn it won't use the correct icon when it fires. Guncode is a confusing mess IMO.